### PR TITLE
Enhancements for PullRequest Resource docs

### PR DESCRIPTION
**Pull request by hrishin**
# Changes

It seems like users may not able able to understand
how to use the `PullRequestResource` with secrete configuration

This patch adds an example TaskRun/Task to demonstrate
how to use the PR Resource.

Fixes https://github.com/tektoncd/pipeline/issues/1275

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

**Comment by sbwsg** [comment]
The example you've added is being executed as part of the YAML tests and appears to be failing because the secret's token isn't valid base64.

Here's the error I'm seeing in the integration test output:

```
I0905 11:07:06.706] Error from server (BadRequest): error when creating "STDIN": Secret in version "v1" cannot be handled as a Secret: v1.Secret.ObjectMeta: v1.ObjectMeta.TypeMeta: Kind: Data: decode base64: illegal base64 data at input byte 6, error found in #10 byte of ...|n_secrete"},"kind":"|..., bigger context ...|Version":"v1","data":{"token":"github_pan_secrete"},"kind":"Secret","metadata":{"annotations":{"kube|...
I0905 11:07:06.718] 2019/09/05 11:07:06 error executing 'kubectl apply': exit status 1
... <snip> ...
I0905 11:17:39.638] ERROR: test pull-request-example=SucceededUnknown but should be succeededtrue
```

**Comment by hrishin** [comment]
@sbwsg ahh Thank you for pointing exact issue (thought so :sweat_smile: ).
Is there any workaround for it? I guess we need to the doc and drop the example.

**Comment by sbwsg** [comment]
It's up to you - if you want this example to be executed as part of the test suite (and therefore verified as working with each PR) then I suggest fixing the base64 string.

Otherwise, adding the example to documentation instead would be the way to go!

**Comment by hrishin** [comment]
@sbwsg I think its better have it into the documentation.

File: docs/resources.md [hunk 1]
```diff
@@ -402,10 +411,12 @@ provider statuses, see
 #### GitHub
 
 The Pull Request resource will look for GitHub OAuth authentication tokens in
-spec secrets with a field name called `authToken`.
+spec secrets with a field name called `githubToken`.
 
 URLs should be of the form: https://github.com/tektoncd/pipelines/pulls/1
 
+[Example GitHub PR task](/examples/taskruns/pull-request.yaml)
```
Comment (sbwsg) [inline docs/resources.md]: This file no longer exists :)

Comment (hrishin) [reply docs/resources.md]: :sweat_smile:

**Comment by hrishin** [comment]
holding it for https://github.com/tektoncd/pipeline/issues/1275#issuecomment-528234881

File: docs/resources.md [hunk 2]
```diff
@@ -370,6 +370,15 @@ spec:
...
       secretName: github-secrets
       secretKey: token
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: github-secrets
+type: Opaque
+data:
+  token: github_personal_access_token_secrete # in base64 encoded value
```
[the remark this answers is not in the document]
Comment (hrishin) [reply docs/resources.md]: :+1:  good catch!

**Pull request merged**